docs/android/gradle.md

```diff
@@ -90,64 +117,53 @@ If the project is configured to prefer settings repositories maven source should
     }
     ```
 
-
-In another case, if `allprojects` is not present in top level `build.gradle` then add it in the top of the file.  
-
-!!! info
-    If your build environment does not allow network access to our servers, 
-    you will be provided with an archive with a local version of PS Plugin. 
-    Unarchive it and put maven packages in an internal artifactory or local maven storage (`~/.m2/repository/`).  
-    
-    Then add internal artifactory or local maven to the project's repositories instead of public PS artifactory:
-    ```
-    repositories {
-        maven {
-            url "https://company.com/artifactory"
-        }
-        mavenLocal() // If local maven is used
-    }
-    ```
+If this block does not exist, you can create it at the top level of your **settings.gradle** file.
 
 
-## 3. Add Product Science plugin to `classpath`
+### Case 2: Project defines module dependencies in an `allprojects` block in the root **build.gradle** file
+Add the repository url to the `allprojects` block in the root **build.gradle** file:
 
 === "Groovy"
     ```groovy title="build.gradle"
-    buildscript {
-        repositories { ... }
-        dependencies {
-            classpath "com.productscience.transformer:transformer-plugin:{{ android_release() }}"
-            classpath "com.productscience.transformer:transformer-instrumentation:{{ android_release() }}"
+    
+    allprojects {
+        repositories {
+            maven {
+                url "https://artifactory.productscience.app/releases"
+            }
         }
     }
-    ...
-    ```
+    ```  
 
 === "Kotlin DSL"
     ```kotlin title="build.gradle.kts"
-    buildscript {
-        repositories { ... }
-        dependencies {
-            classpath("com.productscience.transformer:transformer-plugin:{{ android_release() }}")
-            classpath("com.productscience.transformer:transformer-instrumentation:{{ android_release() }}")
+
+    allprojects {
+        repositories {
+            maven {
+                url = uri("https://artifactory.productscience.app/releases")
+            }
         }
     }
-    ...
     ```
 
-**Please label your build with the PSi Plugin Version from above i.e.**  
-`MyAppPSi0.9.1.apk` 
-**so our AI can learn how its dynamic instrumentation is performing on the build.**
+If this block does not exist, you can create it at the top level of your **build.gradle** file.
+
+
+**Note:** If you use `RepositoriesMode.FAIL_ON_PROJECT_REPOS` of `dependencyResolutionManagement` you 
+may experience a failure when one or more modules define their own repositories in their **build.gradle** files.  
+Switching the mode to `RepositoriesMode.PREFER_SETTINGS` may solve this problem.
+Alternatively, you may add the productscience maven repository url to these module **build.gradle** files.
```

Comment:
I encountered exactly this issue when I used the assistant, and it was quite helpful in resolving it. Good to add it here.
Minor grammar:
Alternatively, you may add the `productscience` maven repository url to these modules` **build.gradle** files.